### ISSUE TITLE
Applications installation wizard

### ISSUE THREAD

**Issue by jsilvestre** [cozy/cozy-home#267]
## Scenarios
```gherkin
Feature:  installation wizard
    The current version of the installation wizard is lackluster.
    This should be the first thing a new user goes through, so it should be a simple, clear, and useful step.

    Scenario: there are no applications installed
        Given the user has no installed applications
        When it reaches the wizard URL
        Then it should see the first screen of the wizard

    Scenario: there are applications installed
        Given the user has application installed
        When it reaches the wizard URL
        Then it should see only the steps for applications that are not installed

    Scenario: the wizard is repeatable
        Given the user has already been through the wizard
        When it reaches the wizard URL
        Then it should start the wizard again

    Scenario: visual indicator
        Given the user is on a step of the wizard
        Then it should see a visual indicator about its progress in the wizard

    Scenario: the user browses a step screen
        Given the user is on a step of the wizard
        Then it should see a question with two choices about an application

    Scenario: the user choses "no" on a step screen
        Given the user is on a step of the wizard
        When it choses "no"
        Then it should go the next step
            And it should not install the application

    Scenario: the user choses "yes" on a step screen
        Given the user is on a step of the wizard
        When it choses "yes"
        Then it should go the next step
            And it should install the application
```

I find hard to script the following information:

## Applications list (in order)
1. Emails
2. Files
3. Contacts (+ sync "hidden")
4. Calendar (+ sync "hidden")

**Comment by jsilvestre** [comment]
Let's try the review process imagined by @frankrousseau, I will do my functional feedback here.

# General
* I thought there would be no need for 1 route / step, but I found myself actually using previous, and I got lost. Thoughts?

# Wizard
* buttons don't have `cursor: pointer` resulting in weird feeling
* you use the orange color for the "positive choice", whereas it's the hover color
* there is no hover color (cf previous point)
=> I suggest you use orange for hover color like in the rest of the application, and you change the "negative choice" button by a text without background and the color of the background.

# Quick tour
I'm still not convinced by this approach vs tourist.js (even more now I can see the result actually). I think it doesn't bing much without context i.e. you show an icon without showing where is this icon on the screen, because you can't, because it's hidden by the modal.

* same UI/UX remarks for buttons
* second step has a 100% width (not the other ones)
* I don't know how I came to that point (probably a bunch or previous/next) but I can see the wizard behind the quick tour modal

**Comment by jsilvestre** [comment]
I found a bug: when hovering the fourth menu buttons, I have this:
![menuhomebug](https://cloud.githubusercontent.com/assets/568111/6076708/3ce893f8-ade7-11e4-810f-aaadbd854f19.png)

**Comment by m4dz** [comment]
> I thought there would be no need for 1 route / step, but I found myself actually using previous, and I got lost. Thoughts?

I asked myself about this point during dev, and I think it's not completely useless: maybe some user will try a back step, so each screen must have its own route

> I suggest you use orange for hover color like in the rest of the application, and you change the "negative choice" button by a text without background and the color of the background.

I agree, let's go.

> second step has a 100% width (not the other ones)

Strange, I'll check why.

**Comment by m4dz** [comment]
> I'm still not convinced by this approach vs tourist.js

Me too, but I think it's going in a good direction. I don't like tourist on its own part, because I find those kind of pooping stuff boxes everywhere totally disrupting, especially because it pollute your screen and you can't easily stop it, nor know how many tips you  must view before finishing the tour.

I think a rich-wizard should be more comfortable, if it produces:
- a simple stop action (actually ESC key, but we can add an optional close button)
- a highlight-target mechanism: each screen could declares one or more targets. Those are highlighted (a-la-tourist.js) when screen is displayed on screen. Those targets are optionals.

Thoughts?

**Comment by jsilvestre** [comment]
Tourist.js lets you have the "step 1 of 2" thing (see first example: http://easelinc.github.io/tourist/). Your points are implementation details, we can do it right with tourist.js.

About what you suggest, I don't understand the difference with what I suggest based on tourist (nb: I say tourist because it's the only lib I know)

**Comment by jsilvestre** [comment]
And don't get me wrong, I'm opinionated, not stubborn! That was a general remark, unrelated to the actual "let's merge this" discussion. What you have done is a very first good step, no matter what the future direction is.

**Comment by m4dz** [comment]
Yep, I've take it as it. I think this step is a first-step before something more powerful.

About the future, I'll make some sketch to illustrates solutions :)

**Comment by jsilvestre** [comment]
There is one more point that is buggy:
> I don't know how I came to that point (probably a bunch or previous/next) but I can see the wizard behind the quick tour modal
It feels weird because I have the wizard again when I'm done with the quick tour.

* steps to reproduce: go through the wizard, then go to the quicktour, click "previous", click "next", go through the quick tour.
* what I see: the wizard modal
* what I should see: apps' icon

**Comment by jsilvestre** [comment]
I just noticed during the demo that you wrote a little tutorial for the synchronization part. I suggest you replace it by just the three links http://cozy.io/mobile/files.html, http://cozy.io/mobile/calendar.html and http://cozy.io/mobile/contacts.html. Reasons: they are very complete step-by-step tutorials, and people always ask us the same questions about it.

**Comment by m4dz** [comment]
OK, I make those few changes ;). Stay tuned!

**Comment by jsilvestre** [comment]
I don't see the the links to the tutorials in the sync screen, can you add them please? Remove the part about davdroid, because it's android-specific, and we support more platform for contacts/calendar sync.

**Comment by m4dz** [comment]
Oh fuck, I forgot those ones… I'll fix that!

**Comment by m4dz** [comment]
Closed by #282
